android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt

```diff
@@ -67,13 +67,17 @@ class AlarmService : Service() {
         val fadeDuration = intent.getDoubleExtra("fadeDuration", 0.0)
         val fullScreenIntent = intent.getBooleanExtra("fullScreenIntent", true)
 
+        // Fetching notificationSettings safely or using default
         val notificationSettingsJson = intent.getStringExtra("notificationSettings")
-        val jsonObject = JSONObject(notificationSettingsJson)
-        val map: MutableMap<String, Any> = mutableMapOf()
-        jsonObject.keys().forEach { key ->
-            map[key] = jsonObject.get(key)
+        val notificationSettings = if (!notificationSettingsJson.isNullOrEmpty()) {
+            val jsonObject = JSONObject(notificationSettingsJson)
+            val map: MutableMap<String, Any> = mutableMapOf()
+            jsonObject.keys().forEach { key -> map[key] = jsonObject.get(key) }
+            NotificationSettings.fromJson(map)
+        } else {
```

Comment:
Maybe remove the else block ? Well notificationSettingsJson is never supposed to be null or empty but if it's the case then we don't want to show a notification.

Reply:
It is empty when users switched from old alarm library versions if I understand correctly.

Reply:
you're right. I realize I have to add backward compatibility with older models in native code. Don't worry I'll add it myself. You can close this PR. Thanks for pointing this issue out.